drivers/include/net/netdev/eth.h

```diff
@@ -29,6 +29,36 @@
 extern "C" {
 #endif
 
+/**
+ * @name    Flags for use in @ref netdev_eth_rx_info_t::flags
+ * @{
+ */
+#define NETDEV_ETH_RX_INFO_FLAG_TIMESTAMP       (0x01)  /**< Timestamp valid */
+/** @} */
+
+/**
+ * @brief   Received frame status information for Ethernet devices
+ */
+typedef struct {
+    /**
+     * @brief   Time of the reception of the start of frame delimiter in
+     *          nanoseconds since epoch
+     */
+    uint64_t timestamp;
+    uint8_t flags;      /**< Flags e.g. used to mark other fields as valid */
+} netdev_eth_rx_info_t;
+
+/**
+ * @brief   Transmitted frame status information for Ethernet devices
+ */
+typedef struct {
+    /**
+     * @brief   Time of the transmission of the start of frame delimiter in
+     *          nanoseconds since epoch
+     */
+    uint64_t timestamp;
+} netdev_eth_tx_info_t;
```

Comment:
None of this looks specific to ethernet - couldn't the RX timestamp be used with any netdev?